File: services/orchestrator/app/tools/firecrawl.py
from typing import Any, Dict

from .mcp import MCPClient


async def autonomous_research(
    mcp_client: MCPClient,
    actor_id: str,
    query: str,
    approval_id: str | None = None,
) -> Dict[str, Any]:
    return await mcp_client.call_tool(
        actor_id=actor_id,
        intent="research_navigation",
        tool_name="firecrawl.agent.search",
        input_payload={"query": query, "depth": 2},
        approval_id=approval_id,
    )

File: src/lib/mcp/firecrawl.ts
import { invokeMcpTool } from "@/lib/mcp/client";

export type FirecrawlAutonomousInput = {
  query: string;
  depth?: number;
};

export async function runFirecrawlAutonomousSearch(
  mcpEndpoint: string,
  actorId: string,
  input: FirecrawlAutonomousInput
) {
  return invokeMcpTool(mcpEndpoint, {
    actorId,
    intent: "research_navigation",
    toolName: "firecrawl.agent.search",
    input
  });
}
